### PR TITLE
Replace Measure instrument by ValueRecorder instrument

### PR DESCRIPTION
**Pull request by jmacd**
This renames "Measure" to "ValueRecorder" and updates some of the doc comments, removing a lot of stale content.

Part of #729.

### CODE REVIEW

File: example/prometheus/main.go [hunk 1]
```diff
@@ -60,11 +60,11 @@ func main() {
 		result.Observe(value, labels...)
 	}
 	_ = metric.Must(meter).RegisterFloat64Observer("ex.com.one", cb,
-		metric.WithDescription("A measure set to 1.0"),
+		metric.WithDescription("A valuerecorder set to 1.0"),
```
Comment (MrAlias) [inline example/prometheus/main.go:63]:
```suggestion
		metric.WithDescription("An observer set to 1.0"),
```

File: example/prometheus/main.go [hunk 2]
```diff
@@ -60,11 +60,11 @@ func main() {
...
 	)
 
-	measureTwo := metric.Must(meter).NewFloat64Measure("ex.com.two")
-	measureThree := metric.Must(meter).NewFloat64Counter("ex.com.three")
+	valuerecorderTwo := metric.Must(meter).NewFloat64ValueRecorder("ex.com.two")
+	valuerecorderThree := metric.Must(meter).NewFloat64Counter("ex.com.three")
```
Comment (MrAlias) [inline example/prometheus/main.go:67]: nit

```suggestion
	valuerecorder := metric.Must(meter).NewFloat64ValueRecorder("ex.com.two")
	counter := metric.Must(meter).NewFloat64Counter("ex.com.three")
```

which would propagate below...

File: sdk/metric/doc.go [hunk 3]
```diff
@@ -13,57 +13,34 @@
...
+behavior through a collection of export interfaces that support
+various export strategies, described below.
+
+The metric.MeterImpl API consists of methods for constructing
+synchronous and asynchronous instruments.  There are two constructors
+per instrument for the two kinds of number (int64, float64).
+
+Synchronous instruments are managed by a sync.Map containing a *record
+with the current state for each synchronous instrument.  A bound
+instrument encapsulates a direct poinnter to the record, allowing
```
Comment (MrAlias) [inline sdk/metric/doc.go:27]:
```suggestion
instrument encapsulates a direct pointer to the record, allowing
```

File: sdk/metric/doc.go [hunk 4]
```diff
@@ -97,21 +69,19 @@ enters the SDK resulting in a new record, and collection context,
...
-pipeline, containing the name, recommended aggregation keys, units,
-description, metric kind (counter or measure), number kind (int64 or
-float64), and whether the instrument has alternate semantics or not (i.e.,
-monotonic=false counter, absolute=false measure).  A Descriptor accompanies
-metric data as it passes through the export pipeline.
+Descriptor is a struct that describes the metric instrument to the
+export pipeline, containing the name, recommended aggregation keys,
+units, description, metric kind, number kind (int64 or float64).  A
+Descriptor accompanies metric data as it passes through the export
+pipeline.
```
Comment (MrAlias) [inline sdk/metric/doc.go:76]: No recommended keys anymore.

```suggestion
export pipeline, containing the name, units, description, metric kind, and
number kind (int64 or float64). A Descriptor accompanies metric data as it
passes through the export pipeline.
```

File: example/prometheus/main.go (hunk 1 above)
Comment (jmacd) [reply example/prometheus/main.go]: Done.

File: example/prometheus/main.go (hunk 2 above)
Comment (jmacd) [reply example/prometheus/main.go]: Done.

File: sdk/metric/doc.go (hunk 4 above)
Comment (jmacd) [reply sdk/metric/doc.go]: Done.

File: sdk/metric/doc.go (hunk 3 above)
Comment (jmacd) [reply sdk/metric/doc.go]: Done.

**Pull request merged**